ForgeAlloy/ForgeAlloyUnity/Assets/ForgeNetworking/Networking/Messaging/ForgeMessageBus.cs

```diff
@@ -117,14 +117,15 @@ private void InterpretWithinContext(object state)
 		{
 			var s = (SocketContainerSynchronizationReadData)state;
 
-			// TODO:  I don't like this type check and if branching in here...
-			bool isServer = _networkMediator.SocketFacade is ISocketServerFacade;
-			if (s.Interpreter.ValidOnClient && !isServer)
-				s.Interpreter.Interpret(_networkMediator, s.Sender, s.Message);
-			else if (s.Interpreter.ValidOnServer && isServer)
+			if (ShouldInterpret(s))
```

Comment:
Also thanks for breaking this into a function! Much more readable inline

Reply:
Been reading through _Clean Code_, trying to put some of into practice.